packages/indicators/src/Builder/ArithmeticBuilder/ArithmeticBuilder.ts

```diff
@@ -80,17 +82,31 @@ export class ArithmeticBuilder extends Builder {
   private getVariables = () => Object.keys(this.config.aggregation);
 
   private fetchAnalytics = async (fetchOptions: FetchOptions) => {
-    const formulaAnalytics = await fetchAnalytics(
-      this.api.getAggregator(),
-      stripFields(this.config.aggregation, Object.keys(this.paramBuildersByCode)),
-      fetchOptions,
-    );
-    const parameterAnalytics = await this.api.buildAnalyticsForBuilders(
-      this.paramBuilders,
-      fetchOptions,
+    const formulaAnalytics = await this.fetchFormulaAnalytics(fetchOptions);
+    const parameterAnalytics = await this.fetchParameterAnalytics(fetchOptions);
+    return [...formulaAnalytics, ...parameterAnalytics];
+  };
+
+  private fetchFormulaAnalytics = async (fetchOptions: FetchOptions) => {
+    const aggregationLisByElement = stripFields(
+      this.config.aggregation,
+      Object.keys(this.paramBuildersByCode),
     );
+    return fetchAnalytics(this.api.getAggregator(), aggregationLisByElement, fetchOptions);
+  };
 
-    return [...formulaAnalytics, ...parameterAnalytics];
+  private fetchParameterAnalytics = async (fetchOptions: FetchOptions) => {
+    const analytics = await this.api.buildAnalyticsForBuilders(this.paramBuilders, fetchOptions);
+    const analyticsByElement = groupBy(analytics, 'dataElement');
+
+    return Object.entries(analyticsByElement)
+      .map(([element, analyticsForElement]) => {
+        const aggregationList = this.config.aggregation[element];
+        return this.api
+          .getAggregator()
+          .aggregateAnalytics(analyticsForElement, aggregationList, fetchOptions.period);
```

Comment:
This is decoupled fetching/aggregation of analytics in practice. It's not required by any existing use case (it isn't supported in `dev`, after all), but it's required by one new [indicator](https://github.com/beyondessential/tupaia/blob/1779-optimise-data-fetching-for-indicators/packages/indicators/src/__tests__/__integration/fixtures/arithmeticAnalyticsFixtures.ts#L122) in the integration tests

Reply:
Yeah nice 👍